File: packages/common/erp-common-ms-0-js/src/servers/graphql/ms/3/estate.ts
import { IEstate, ICreateEstate, IUpdateEstate, IDeleteEstate, ISearchEstate } from '@gmahechas/erp-common';
import { axiosClient, env } from '@gmahechas/erp-common-ms-utils-js';

export const createOneEstate = async (data: ICreateEstate): Promise<IEstate> => {
	const { data: response } = await axiosClient(env.ms!.three!.command!.url!).post<{ data: IEstate }>('/rest/v1/3/estate/create/one', data);
	return response.data;
};

export const updateOneEstate = async (data: IUpdateEstate): Promise<IEstate> => {
	const { data: response } = await axiosClient(env.ms!.three!.command!.url!).put<{ data: IEstate }>('/rest/v1/3/estate/update/one', data);
	return response.data;
};

export const deleteOneEstate = async (data: IDeleteEstate): Promise<IEstate> => {
	const { data: response } = await axiosClient(env.ms!.three!.command!.url!).delete<{ data: IEstate }>('/rest/v1/3/estate/delete/one', { data });
	return response.data;
};

export const searchOneEstate = async (data: Partial<ISearchEstate>): Promise<IEstate> => {
	const { data: response } = await axiosClient(env.ms!.three!.command!.url!).post<{ data: IEstate }>('/rest/v1/3/estate/search/one', data);
	return response.data;
};

export const searchManyEstate = async (data: Partial<ISearchEstate>[]): Promise<IEstate[]> => {
	const { data: response } = await axiosClient(env.ms!.three!.command!.url!).post<{ data: IEstate[] }>('/rest/v1/3/estate/search/many', data);
	return response.data;
};

File: packages/3/ms-3-gingo/routes/v1/estate.go
package routes

import (
	ginGoUtils "github.com/gmahechas/erp-common-gin-go/utils"
)

func InitEstateRoutes(router *ginGoUtils.GinRouterGroup) {
	routes := router.Group("/estate")
	{
		routes.GET("/create/one", ginGoUtils.GinHandlerFunc(createOneEstate))
		routes.GET("/update/one", ginGoUtils.GinHandlerFunc(createOneEstate))
		routes.GET("/delete/one", ginGoUtils.GinHandlerFunc(deleteOneEstate))
		routes.GET("/search/one", ginGoUtils.GinHandlerFunc(searchOneEstate))
		routes.GET("/search/many", ginGoUtils.GinHandlerFunc(searchManyEstate))
	}
}

func createOneEstate(context *ginGoUtils.GinContext) {
	context.String(200, "createOneEstate !!!")
}

func updateOneEstate(context *ginGoUtils.GinContext) {
	context.String(200, "updateOneEstate !!!")
}

func deleteOneEstate(context *ginGoUtils.GinContext) {
	context.String(200, "deleteOneEstate !!!")
}

func searchOneEstate(context *ginGoUtils.GinContext) {
	context.String(200, "searchOneEstate !!!")
}

func searchManyEstate(context *ginGoUtils.GinContext) {
	context.String(200, "searchManyEstate !!!")
}
